### PR TITLE
Update Go version to 1.16 in go.mod, drop < 1.16 from versions tested, update README.md fixes #431

### LINKED ISSUE #431: Follow the lead of the Go team and support the last two major versions of Go
This issue was created in response [this suggestion](https://github.com/ably/ably-go/pull/426#issuecomment-953603474) by @Rosalita and [this reply](https://github.com/ably/ably-go/pull/426#issuecomment-954160872) by @lmars.

Concrete steps (duplicated from previously referenced comment):

1. Set `go 1.16` in `go.mod` (since that's now our minimum supported version)
2. Drop versions `< 1.16` from the list of versions we test against
3. Document our policy in the README
4. Sync our schedule with that of the Go team (so when `1.18` is released, bump to `go 1.17` in `go.mod` + switch to testing `1.17 + 1.18` in CI)

Relates to #344.

┆Sprint: Backlog

### PR DESCRIPTION
**Pull request by Rosalita**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ This is a Go client library for Ably.
 ~ $ go get -u github.com/ably/ably-go/ably
 ```
 
+## Supported Versions of Go
```
Comment (QuintinWillison) [inline README.md:15]: For consistency with other headers in this file, could you please add vertical whitespace below this line. I recognise it doesn't affect the rendering in GitHub's UI, however it does affect those who wish to read it raw as well as those maintaining the file.

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Just one minor tweak, please. 😄 

I've also added @tomkirbygreen as a reviewer. He was away last week but has been providing oversight in this repository so I would appreciate his input here too.

**Comment by QuintinWillison** [comment]
Thanks, @Rosalita ... because you've force pushed the branch underlying this pull request I'm going to re-request review from @lmars.

(On Ably SDK Team, at least, we prefer to treat pull requests as ongoing stories and, therefore, will simply add commits to the underlying branch to reflect changes made in response to comments. I acknowledge this policy is not - yet - published anywhere that you would have access to. I am fixing that soon. 😄)

**Comment by Rosalita** [comment]
> Thanks, @Rosalita ... because you've force pushed the branch underlying this pull request I'm going to re-request review from @lmars.
> 
> (On Ably SDK Team, at least, we prefer to treat pull requests as ongoing stories and, therefore, will simply add commits to the underlying branch to reflect changes made in response to comments. I acknowledge this policy is not - yet - published anywhere that you would have access to. I am fixing that soon. 😄)

No worries @QuintinWillison I used `git commit --amend` here followed by `git push --force-with-lease` to avoid unnecessary noise in the main branch history. On my current team they all prefer to keep the commit history in main as clean as possible with each change squashed into a single commit. With enforced conventional commit specification (https://www.conventionalcommits.org/en/v1.0.0/) using multi-line message if necessary to describe the change. This is so that release notes can be generated directly from the repo. So I'm used to squashing and amending to keep each merge to main as small and readable as possible. If you are adding documentation regarding best practice in git, if you haven't seen it already, you may want to take a look at the Go contribution guide, where they have a whole section just for commit messages https://golang.org/doc/contribute

**Review by lmars — APPROVED** [review]

**Comment by QuintinWillison** [comment]
Sorry, it's been a busy day, so I've only just got around to looking at this again.

In respect of clean and/or linear commit history, including enforcing commits to a common specification: We have other teams at Ably, working on internal codebases, that do adopt this kind of approach - including regular rebase on feature branches and encouraged use of squash and/or rebase on landing pull requests to `main` branch. For our open source SDK repositories my preference is to treat pull requests more as stories, retaining as much of the history as possible. This includes always landing pull requests to the target branch with merge commits, and updating the branches underlying pull requests with merge commits from their target branch. This has been proven, for this team, to help with discoverability later on when we're doing archeology to discover 'reasons' for things. It's also more open, generally aligning better with our company values. Finally, not forgetting that we never know who may have pulled in a feature branch to their local machine to inspect or work on it - so keeping rebase out of the mix makes that more fluid for everyone collaborating.

I am always open to evolving our processes so this could be tweaked in future. I also recognise that what I've just captured above is still on my backlog to actually write down somewhere succinctly and open / public (some of it's in internal Wiki and what's at [engineering.ably.com](https://engineering.ably.com/) does not reflect what we do on SDK Team). It's on the immediate radar - we're always wanting to improve, but there are only ever so many hours in the day. 😅 

In terms of the structure of commit messages specifically, I am of the strong opinion that I would rather leave it up to individual contributors to decide the style of commit message they prefer. Line length limits, insistence on first lines being summaries, fancy prefixes / suffixes, and more are often impediments to simply getting the work done. There's always a balance in these things, I recognise that. But, genuinely, day to day I've never experienced - at least for SDK Team work at Ably - the commit messages that individuals write to have been something that causes problems or otherwise slows people down downstream. There are many things that surprise (and sometimes disappoint) me about the Go ecosystem, including oddly opinionated / borderline-obstinate tooling ... so, I'll admit a section in their contributing guideline docs entitled "Good commit messages" doesn't surprise me, though it does feel out of scope as a thing to insist contributors should align with. I feel that the barrier for entry for any contributor needs to be kept low. Artificial, subjective limits around commit messages seem unnecessary and arbitrary to me.

**Review by QuintinWillison — APPROVED** [review]

**Comment by Rosalita** [comment]
@QuintinWillison thank you for approval, sadly Github tells me I'm "not authorized to merge this pull request" so I will let the team merge at their discretion.

Rosie.

**Comment by QuintinWillison** [comment]
@tomkirbygreen once you've reviewed, and assuming you're happy to approve, then feel free to land this pull request.

**Review by tomkirbygreen — APPROVED** [review]

**Pull request merged**